addons/website/static/src/snippets/s_dynamic_snippet_carousel/000.xml

```diff
@@ -1,36 +1,59 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <templates xml:space="preserve">
+    <t t-name="website.s_dynamic_snippet.carousel.arrows">
+        <a t-attf-href="##{uniqueId}" t-attf-class="o_w_carousel_control carousel-control-prev" data-slide="prev" role="button" aria-label="Previous" title="Previous">
+            <span class="fa fa-chevron-circle-left fa-2x"/>
+            <span class="sr-only">Previous</span>
+        </a>
+        <a t-attf-href="##{uniqueId}" t-attf-class="o_w_carousel_control carousel-control-next" data-slide="next" role="button" aria-label="Next" title="Next">
+            <span class="fa fa-chevron-circle-right fa-2x"/>
+            <span class="sr-only">Next</span>
+        </a>
+    </t>
     <t t-name="website.s_dynamic_snippet.carousel">
         <div t-att-id="uniqueId" class="carousel slide" t-att-data-interval="interval">
             <!-- Content -->
+            <t t-set="slideSize" t-value="chunkSize * rowPerSlide" />
             <t t-set="colClass" t-value="'d-flex flex-grow-0 flex-shrink-0 col-' + Math.trunc(12 / chunkSize).toString()"/>
-            <t t-set="slideIndexGenerator" t-value="Array.from(Array(Math.ceil(data.length/chunkSize)).keys())"/>
-            <t t-set="itemIndexGenerator" t-value="Array.from(Array(chunkSize).keys())"/>
+            <t t-set="slideIndexGenerator" t-value="Array.from(Array(Math.ceil(data.length/slideSize)).keys())"/>
+            <t t-set="rowIndexGenerator" t-value="Array.from(Array(rowPerSlide).keys())"/>
+            <t t-set="colIndexGenerator" t-value="Array.from(Array(chunkSize).keys())"/>
             <div class="carousel-inner row w-100 mx-auto" role="listbox">
                 <t t-foreach="slideIndexGenerator" t-as="slideIndex">
-                    <div t-attf-class="carousel-item #{slideIndex_first ? 'active' : ''}">
-                        <div class="row">
-                            <t t-foreach="itemIndexGenerator" t-as="itemIndex">
-                                <t t-if="(slideIndex * chunkSize + itemIndex) &lt; data.length">
-                                    <div t-attf-class="#{colClass}">
-                                        <t t-out="data[slideIndex * chunkSize + itemIndex]"/>
-                                    </div>
-                                </t>
+                    <t t-set="slideOffset" t-value="slideIndex * slideSize"/>
+                    <div t-attf-class="carousel-item #{extraClasses} #{slideIndex_first ? 'active' : ''}">
```

Comment:
If find no other reference to `slideIndex_first`. Maybe `slideIndex == 0` would be an equivalent condition to what was intended ?

Reply:
I'm not sure I understand this. As per [Odoo Docs](https://www.odoo.com/documentation/15.0/developer/reference/frontend/qweb.html#loops):
> $as_first
whether the current item is the first of the iteration (equivalent to $as_index == 0)

Maybe this doesn't work in QWeb JS?

Reply:
Oops, I had forgotten about this, ignore my comment.
QWeb.js seems to support it:
```js
context[val_first] = j == 0;
```